test/features/recognize.js

```diff
@@ -0,0 +1,338 @@
+    controller = new MockController({});
+
+    controller.bot.api.users.info
+      .withArgs({ user: "Giver" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "Giver",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "Receiver" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "Receiver",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "BotUser" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "BotUser",
+          tz: "America/Los_Angeles",
+          is_bot: true,
+          is_restricted: false,
+        },
+      })
+      .withArgs({ user: "GuestUser" })
+      .resolves({
+        ok: true,
+        user: {
+          id: "GuestUser",
+          tz: "America/Los_Angeles",
+          is_bot: false,
+          is_restricted: true,
+        },
+      })
+      .withArgs({ user: "NotARealUser" })
+      .resolves({
+        ok: false,
+        error: "user_not_found",
+      });
+
+    await recognizeFeature(controller);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("a recognition message", () => {
+    it("should update database on valid recognition", async () => {
+      const giveRecognition = sinon
+        .stub(recognition, "giveRecognition")
+        .resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@Receiver> Test Test Test Test Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      expect(giveRecognition.called).to.be.true;
+    });
+
+    it("should parse out supplied message tags", async () => {
+      const giveRecognition = sinon
+        .stub(recognition, "giveRecognition")
+        .resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@Receiver> Test Test Test Test #Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      expect(giveRecognition.args[0][4]).to.deep.equal(["Test"]);
+    });
+
+    it("should handle Slack API errors", async () => {
+      sinon.stub(recognition, "giveRecognition").resolves("");
+      sinon.stub(recognition, "countRecognitionsReceived").resolves(1);
+      sinon.stub(balance, "dailyGratitudeRemaining").resolves(5);
+
+      await controller.userInput({
+        text: ":fistbump: <@NotARealUser> Test Test Test Test Test",
+        user: "Giver",
+        channel: "SomeChannel",
+      });
+
+      // TODO: Test something
+    });
+
+    /*
+     * TODO: Need to figure out the correct way to mock config
```

Comment:
For `config`, one possibility is to overwrite the value and then set it back after the test. Not sure if Sinon would be able to mock the property getter

Reply:
I think I tried this, both with Sinon and without, and it didn't properly set the value. My guess was that the variable was already scoped to recognize.js by the time the change occurred, but I don't have a great understanding of the execution order when files are required.

I'll play around with it a bit more to see if I can manage something.

Reply:
It seems that this probably has to due with the ordering of commands that I mentioned before. The variables from config.js are set to variables at the very top of the file, and afterwards don't respond to config.js being mocked.

If I simply reference the variable as config.usersExemptFromMaximum the mock works. I don't mind the change, although it does seem like an odd change to have to make.